nemo/deploy/nlp/trtllm_pytorch_deployable.py

```diff
@@ -0,0 +1,221 @@
+
+        This method processes input prompts through the loaded model and
+        generates text according to the specified parameters.
+
+        Args:
+            prompts: List of input prompts
+            max_length: Maximum number of tokens to generate. Defaults to 256.
+            temperature: Sampling temperature. Defaults to None.
+            top_k: Number of highest probability tokens to consider. Defaults to None.
+            top_p: Cumulative probability threshold for token sampling. Defaults to None.
+
+        Returns:
+            List[str]: A list of generated texts, one for each input prompt.
+
+        Raises:
+            RuntimeError: If the model is not initialized.
+        """
+        if not self.model:
+            raise RuntimeError("Model is not initialized")
+
+        sampling_params = SamplingParams(
+            max_tokens=max_length,
+            temperature=temperature,
+            top_k=top_k,
+            top_p=top_p,
+        )
+
+        outputs = self.model.generate(
+            inputs=prompts,
+            sampling_params=sampling_params,
+        )
+
+        return [output.outputs[0].text for output in outputs]
+
+    def generate_other_ranks(self):
+        """
+        Generate function for ranks other than the rank 0.
+        """
+        while True:
+            message = torch.empty(1, dtype=torch.long, device="cuda")
+            torch.distributed.broadcast(message, src=0)
+            if message == 0:
+                prompts = broadcast_list(data=[None], src=0)
+                temperature, top_k, top_p, max_length = broadcast_list(data=[None], src=0)
+
+                self.generate(
+                    prompts=prompts,
+                    temperature=temperature,
+                    top_k=top_k,
+                    top_p=top_p,
+                    max_length=max_length,
+                )
+            else:
+                return
+
+    @property
+    def get_triton_input(self):
+        inputs = (
+            Tensor(name="prompts", shape=(-1,), dtype=bytes),
+            Tensor(name="max_length", shape=(-1,), dtype=np.int_, optional=True),
+            Tensor(name="max_batch_size", shape=(-1,), dtype=np.int_, optional=True),
+            Tensor(name="top_k", shape=(-1,), dtype=np.int_, optional=True),
+            Tensor(name="top_p", shape=(-1,), dtype=np.single, optional=True),
+            Tensor(name="temperature", shape=(-1,), dtype=np.single, optional=True),
+            Tensor(name="random_seed", shape=(-1,), dtype=np.int_, optional=True),
+        )
+        return inputs
+
+    @property
+    def get_triton_output(self):
+        return (Tensor(name="sentences", shape=(-1,), dtype=bytes),)
+
+    @batch
+    def triton_infer_fn(self, **inputs: np.ndarray):
+        output_infer = {}
+
+        try:
+            prompts = str_ndarray2list(inputs.pop("prompts"))
+            temperature = inputs.pop("temperature")[0][0] if "temperature" in inputs else None
```

Comment:
Could you please use `first_value` decorator here for consistency/simplicity? See https://triton-inference-server.github.io/pytriton/0.5.14/inference_callables/decorators/#first-value and the recent update in https://github.com/NVIDIA/NeMo/pull/12773